Automattic/newspack-newsletters
# Premium newsletters poc

### All Submissions:






### Changes proposed in this Pull Request:



Implements the proof of concept to support Premium Lists tied to Woocommerce Membership plans

### How to test the changes in this Pull Request:

1. Create a Subscription List via the wp-admin panel (that will create e Tag or Mailchimp group in the ESP)
2. Activate Woocommerce Membership and create a Memebership plan
3. In the "Restrict content" tab, select "Subscription Lists" and add the List you just created as the restricted content
4. Add a Newsletters Subscription Form block to a page and select all available lists, including the premium one
5. As an anonymous reader, confirm you don't see the premium list in the block
6. Perform the action required to be granted the membership (register or purchase a product)
7. Confirm that now you see the list in the block
8. Confirm the in ESP dashboard that the user was automatically added to the list
9. Verify your account and confirm you are able to manage your subscription to the premium list in My Account > Newsletters
10. Cancel the membership and confirm the user was removed from the list in the ESP dashboard


### Other information:
